Lib/typing.py

```diff
@@ -695,6 +701,15 @@ def Union(self, parameters):
         return _UnionGenericAlias(self, parameters, name="Optional")
     return _UnionGenericAlias(self, parameters)
 
+def _make_union(left, right):
+    """Used from the C implementation of TypeVar.
+
+    TypeVar.__or__ calls this instead of returning types.UnionType
+    because we want to allow unions between TypeVars and strings
+    (forward references.)
+    """
+    return Union[left, right]
```

Comment:
I guess this means that `TypeVar.__or__` will be a fair bit slower than `type.__or__`? That feels slightly unfortunate. But I agree that backwards compatibility is more important here.

Reply:
Yes, we can probably reimplement more of it in C later though.

Reply:
Also, we’ll eventually have lazy rival with PEP 649, right?

Reply:
Yeah, it probably won't matter unless you're doing old-style type alias, e.g.

```py
T = TypeVar("T")
X = T | int
```

Though FWIW, I expect many users will continue to use old-style type aliases for many years to come, in order to maintain compatibility with older Python versions

Reply:
But isn’t that an anti pattern? That X becomes a generic type, but it’s easy to miss that and believe it can be used as a shorthand for T|int.

Reply:
I don't think of generic type aliases as an anti-pattern! Mypy documents them as a feature, and that's how I think of them as well. If you use `--disallow-any-generics` (or the equivalent for pyright/pytype/pyre/etc), the type checker will emit an error if you forget to parameterise the alias when using it in an annotation

Reply:
We use generic type aliases a fair bit in typeshed, so it's news to me if they're considered an anti-pattern :-)

Reply:
I don't think generic aliases in general are an antipattern, but it's probably true that useful generic aliases will rarely involve a union with a TypeVar.